src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -67,6 +67,113 @@ internal class FeatureManagementKeyValueAdapter : IKeyValueAdapter
                 keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}", false.ToString()));
             }
 
+            if (featureFlag.Variants != null)
+            {
+                for (int i = 0; i < featureFlag.Variants.Count; i++)
+                {
+                    FeatureVariant featureVariant = featureFlag.Variants[i];
+
+                    keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.Variants}:{i}:Name", featureVariant.Name));
```

Comment:
How about assigning `{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.Variants}:{i}` for reuse in the key names.